File: Core/Entities/OrderAggregate/Order.cs


namespace Core.Entities.OrderAggregate
{
    public class Order : BaseEntity
    {
        public Order()
        {
            
        }
        public Order(IReadOnlyList<OrderItem> OrderItems,
        string email, Address shipToAddress, DeliveryMethod deliveryMethod, decimal subtotal)
        {
            this.BuyerEmail = email;
            this.ShipToAddress = shipToAddress;
            this.DeliveryMethod = deliveryMethod;
            this.Subtotal = subtotal;
            this.OrderItems=OrderItems;

        } 
        public string BuyerEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        public Address ShipToAddress { get; set; }

        public IReadOnlyList<OrderItem> OrderItems { get; set; }

        public DeliveryMethod DeliveryMethod { get; set; }

        public decimal Subtotal { get; set; }

        public OrderStatus Status { get; set; } = OrderStatus.Pending;
        //for stripe payment 
        public string PaymentIntentId { get; set; }

        //automapper reads the GET and map to Total property
        public decimal GetTotal()
        {

            return Subtotal + DeliveryMethod.Price;
        }
    }
}

File: ClientAPP/src/app/shared/Models/order.ts
import { Address } from './User';

export interface Order {
  id: number;
  buyerEmail: string;
  orderDate: string;
  shipToAddress: Address;
  orderItems: OrderItem[];
  deliveryMethod: string;
  shipingPrice: number;
  subtotal: number;
  total: number;
  status: string;
}

export interface OrdertoCreate {
  basketId: string;
  ShipToAddress: Address;
  DeliveryMethodId: number;
}

export interface OrderItem {
  productId: number;
  productName: string;
  pictureUrl: string;
  price: number;
  quantity: number;
}
